### PR TITLE
Delete collection rewrote, dynamic type mapping resolving

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Part of #371.

@nikitin-da PTAL

### CODE REVIEW

File: storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operation/delete/PreparedDeleteCollectionOfObjectsTest.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+        verify(storIOSQLite).delete();
+        verify(storIOSQLite).internal();
+        verify(internal).typeMapping(TestItem.class);
+        verify(internal, never()).delete(any(DeleteQuery.class));
+        verifyNoMoreInteractions(storIOSQLite, internal);
+    }
+
+    @Test
+    public void shouldThrowExceptionIfNoTypeMappingWasFoundWithoutTransactionWithoutAffectingDbObservable() {
```
Comment (nikitin-da) [inline storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operation/delete/PreparedDeleteCollectionOfObjectsTest.java]: Wow such name)

**Comment by nikitin-da** [comment]
Great! But Travis didn't say what's wrong.

**Comment by artem-zinnatullin** [comment]
Because 10k logs limit...

**Comment by artem-zinnatullin** [comment]
So, let's try to rebuild it on TeamCity.

**Comment by artem-zinnatullin** [comment]
Okay, @nikitin-da JFYI open/reopen does not trigger build.

**Comment by nikitin-da** [comment]
Ben would upset)

**Comment by artem-zinnatullin** [comment]
Hehe

**Comment by artem-zinnatullin** [comment]
Yes! Travis build it!

**Pull request merged**